test-engineering/contract-tests/client/tests/models.py

```diff
@@ -21,6 +21,8 @@ class Request(BaseModel):
     method: str
     path: str
     headers: List[Header] = []
+    # Delay is optional, providing time for data refresh
+    delay: Optional[float] = 0
```

Comment:
Normally, optional fields will use `None` as the default value. Did you want this field to be a float with 0 as the default instead?

Reply:
@ncloudioj yes! If the `None` type is passed to `time.sleep()` the function will crash, so I wanted to make sure we strictly set it to 0. Given that optional fields use `None` as a default, do you see a problem with this implementation that overrides it?

Reply:
I think if delay is optional, you could do:

```
if (delay := step.request.delay) is not None:
    time.sleep(delay)
```

Reply:
Goodness @ncloudioj ... the walrus operator in the wild! Isn't this getting fancy.

Reply:
Haha, many also dislike it. I think it's pretty handy.